### PR TITLE
HCAP-1164: Select Site modal

### PR DESCRIPTION
**Pull request by alva-fresh**
https://freshworks.atlassian.net/browse/HCAP-1164
Added a new modal for Participant Table. This modal appears when somebody is trying to Engage a participant
made some updates to `DialogTitle` as well - let me know what you think

This is a subtask for HCAP-1107

Adding some updates to `Prospecting Confirmation` as well to match designs
Screen Shot 2022-04-19 at 5 17 35 PM

https://www.figma.com/proto/35c5C0DyTEOxJ4N4DIaADL/HCAP-Design?node-id=2002%3A19657&viewport=241%2C48%2C0.36&scaling=min-zoom&starting-point-node-id=2002%3A19657

Screen Shot 2022-04-19 at 5 14 45 PM
Screen Shot 2022-04-19 at 5 14 49 PM

_Note_: this is for single participant only

### CODE REVIEW

File: client/src/components/modal-forms/SelectProspectingSiteForm.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          <Typography className={classes.formLabel} variant='subtitle2'>
+            Please select site(s) this participant is prospecting for
+          </Typography>
+          <FastField
+            name='participantSites'
+            component={RenderMultiSelectField}
+            placeholder='Select Site'
+            // TODO: get the list of options
+            options={[
+              { value: 'opt1', label: 'option 1' },
```
Comment (fwpushan) [inline client/src/components/modal-forms/SelectProspectingSiteForm.js:43]: get value from auth-context.

File: client/src/constants/validation.js [hunk 2]
```diff
@@ -580,6 +580,13 @@ export const RejectedFormSchema = yup
       ),
   });
 
+export const SitesSchema = yup.object().shape({
```
Comment (fwpushan) [inline client/src/constants/validation.js:583]: You need to rebase with master .... validation.js is refactored.

**Review by fwpushan — CHANGES_REQUESTED** [review]
Good job. Please check my comments. I also added some tech-notes in ticket (sub-task)

File: client/src/components/modal-forms/SelectProspectingSiteForm.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  return (
+    <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={onSubmit}>
+      {({ submitForm }) => (
+        <FormikForm>
+          <Typography className={classes.formLabel} variant='subtitle2'>
+            Please select site(s) this participant is prospecting for
+          </Typography>
+          <FastField
+            name='participantSites'
```
Comment (fwpushan) [inline client/src/components/modal-forms/SelectProspectingSiteForm.js:38]: `prospectingSites` is more appropriate here.

Comment (fwpushan) [inline client/src/components/modal-forms/SelectProspectingSiteForm.js:38]:
```suggestion
            name='prospectingSites'
```

Comment (alva-fresh) [reply client/src/components/modal-forms/SelectProspectingSiteForm.js:38]: Good point!

**Review by fwpushan — CHANGES_REQUESTED** [review]
All good just one change recommendation.

**Review by fwpushan — APPROVED** [review]

**Pull request merged**